From 1dd69fe9a77ed1a2d1cc839f71135b57f7ffb9f4 Mon Sep 17 00:00:00 2001
From: Niels De Graef <nielsdegraef@gmail.com>
Date: Wed, 16 Sep 2026 19:00:34 +0200
Subject: [PATCH] daemon: Fix double unref of session

Looks like there was an overlap in the leak fixes that I managed to look
over after I rebased one of them, so we're crashing due to a double
free.

Fixes: https://gitlab.gnome.org/GNOME/gnome-keyring/-/work_items/197
---
 daemon/dbus/gkd-secret-session.c | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/daemon/dbus/gkd-secret-session.c b/daemon/dbus/gkd-secret-session.c
index 16bb368d..6799b4ce 100644
--- a/daemon/dbus/gkd-secret-session.c
+++ b/daemon/dbus/gkd-secret-session.c
@@ -609,13 +609,6 @@ gkd_secret_session_get_item_secret (GkdSecretSession *self, GckObject *item,
 					   NULL, &error);
 	g_object_unref (session);
 
-	/*
-	 * gck_object_get_session() returns a new reference, so drop it again.
-	 * Not doing so leaks a GckSession on every GetSecret call, which in turn
-	 * keeps the client's PKCS#11 session alive after the client is gone.
-	 */
-	g_object_unref (session);
-
 	if (error != NULL) {
 		if (g_error_matches (error, GCK_ERROR, CKR_USER_NOT_LOGGED_IN)) {
 			g_set_error_literal (error_out, GKD_SECRET_ERROR,
-- 
GitLab

